node/src/components/consensus/highway_core/state.rs

```diff
@@ -788,9 +788,20 @@ impl<C: Context> State<C> {
 
     /// Returns `true` if the `bhash` is a block that can have no children.
```

Comment:
Now that it's getting complicated it would be good to document it here. Maybe something like:
```suggestion
    /// Returns `true` if the `bhash` is a block that can have no children.
    ///
    /// A block is considered terminal if it has both the configured minimum height and minimum
    /// timestamp, and is at least one maximum round length after the first block. The latter is
    /// required because validators with panorama entry `None` in the last block are reported as
    /// inactive, so they need at least one round to have a chance to create a unit.
    ///
    /// Note that a malicious block proposer could still choose to ignore units and cause someone
    /// to get reported as inactive, so it is recommended to configure a sufficient minimum height.
```